test/foundry/dapp/Application.t.sol

```diff
@@ -523,15 +701,38 @@ contract ApplicationTest is ERC165Test {
         _appOwner = _newAddr();
         _recipient = _newAddr();
         _tokenOwner = _newAddr();
+        _ens = ENS(_newAddr());
+        _resolver = AddrResolver(_newAddr());
         _interfaceIds.push(type(IApplication).interfaceId);
         _interfaceIds.push(type(IERC721Receiver).interfaceId);
         _interfaceIds.push(type(IERC1155Receiver).interfaceId);
+        for (uint256 i; i < 7; ++i) {
+            _tokenIds.push(i);
+            _initialSupplies.push(_initialSupply);
+            _transferAmounts.push(
+                bound(uint256(keccak256(abi.encode(i))), 0, _initialSupply)
+            );
+        }
+        if (_transferAmounts[0] == 0) {
+            // make the first transfer amount non-zero for easier tests
+            _transferAmounts[0] = 1;
+        }
```

Comment:
I want to allow 0 `transferAmount` for other tokens. But keeping the first one non-zero helps with the revert case when the app balance is 0, see [here](https://github.com/cartesi/rollups-contracts/blob/43b13e38100a860f528723565d2f0bcfa4402ec3/test/foundry/dapp/Application.t.sol#L635). If we don't make the first transfer amount non-zero,  then it will not be the first token that reverts, it could be the 2nd or 3rd or 4th, which means we need a loop to know which token we expect to revert. Or if all transfer amounts are 0, then no revert will happen.
So I think it's easier to just make the first amount non-zero.

Reply:
Given that transfer amounts are generated randomly, it is very unlikely that any one of them is zero. But if you want to be extra certain, you can just adjust the minimum transfer amount to be 1 rather than 0, [here](https://github.com/cartesi/rollups-contracts/blob/43b13e38100a860f528723565d2f0bcfa4402ec3/test/foundry/dapp/Application.t.sol#L712-L714).

Reply:
Yes, we could do that too.